ironfish/src/testReporter/testReporter.ts

```diff
@@ -53,26 +52,64 @@ export default class TestReporter implements CustomReporter {
     const writeStream = createWriteStream(
       `${this.globalConfig.rootDir}/${this.reporterConfig.outputDirectory}/${testFileName}.csv`,
     )
-    const stream = format({ headers: true })
-    stream.pipe(writeStream)
 
     // filter console log
     const consoleOutputs = testResult.console?.filter((output) => output.type === 'log')
 
+    // annotation headers for influx data
+    let datatypeHeader = '#datatype,string,long,string,string,string,dateTime:RFC3339'
+    let groupHeader = '#group,false,false,true,true,true,false'
+    let defaultHeader = '#default,_result,,,,,'
+    if (consoleOutputs && consoleOutputs[0]) {
+      const entries = consoleOutputs[0].message.split(',')
+      entries.forEach((input) => {
+        const entry = input.split(':')
+        const value = entry[1]
+        if (value) {
+          const trim = value.replace(/[^\d.-]/g, '')
+          if (trim.includes('.')) {
+            datatypeHeader = datatypeHeader.concat(',double')
+          } else {
+            datatypeHeader = datatypeHeader.concat(',long')
+          }
+          groupHeader = groupHeader.concat(',true')
+          defaultHeader = defaultHeader.concat(',')
+        }
+      })
+    }
+
+    writeStream.write(groupHeader + '\n')
+    writeStream.write(datatypeHeader + '\n')
+    writeStream.write(defaultHeader + '\n')
+
+    const stream = format({ headers: true })
+    stream.pipe(writeStream)
+
     testResult.testResults.forEach((result, i) => {
       const row: Record<string, string> = {
-        name: result.title,
-        duration: String(result.duration),
+        '': '',
+        result: '_result',
```

Comment:
Did you figure out what these two things mean `table` and `result` columns? I was trying to look into the influx docs but its kind of unclear https://docs.influxdata.com/influxdb/cloud-serverless/reference/syntax/annotated-csv/#columns.

Specifically was wondering about the table column because I thought that was related to cardinality. But looking at the results in influx it looks like multiple rows can be in the same table even if their grouped fields are different.

Reply:
Yeah I could not figure out the table and result columns exact purpose. But they are not related to cardinality, these data does not produce random tags.
I followed the sample csv file from https://influx-testdata.s3.amazonaws.com/air-sensor-data-annotated.csv on the page https://us-east-1-1.aws.cloud2.influxdata.com/orgs/fdcfe96f6c31245a/new-user-setup/cli

The sample data from their latest annotated csv does not work for our influxDB, it will fail with parsing error. I suspect we are using an older version of influxDB which does not support the latest annotation.

Reply:
Discussed this and don't think this should impact cardinality. It does seem necessary to avoid formatting errors from influx on import